odiglet/pkg/ebpf/manager.go

```diff
@@ -0,0 +1,415 @@
+	odigosv1 "github.com/odigos-io/odigos/api/odigos/v1alpha1"
+	"github.com/odigos-io/odigos/common"
+	"github.com/odigos-io/odigos/k8sutils/pkg/consts"
+	odgiosK8s "github.com/odigos-io/odigos/k8sutils/pkg/container"
+	instance "github.com/odigos-io/odigos/k8sutils/pkg/instrumentation_instance"
+	workloadUtils "github.com/odigos-io/odigos/k8sutils/pkg/workload"
+	"github.com/odigos-io/odigos/odiglet/pkg/detector"
+	"github.com/odigos-io/odigos/odiglet/pkg/kube/utils"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	ErrNoInstrumentationFactory = errors.New("no ebpf factory found")
+)
+
+type errRequiredEnvVarNotFound struct {
+	envVarName string
+}
+
+func (e *errRequiredEnvVarNotFound) Error() string {
+	return fmt.Sprintf("required environment variable not found: %s", e.envVarName)
+}
+
+var _ error = &errRequiredEnvVarNotFound{}
+
+var (
+	errContainerNameNotReported = &errRequiredEnvVarNotFound{envVarName: consts.OdigosEnvVarContainerName}
+	errPodNameNotReported       = &errRequiredEnvVarNotFound{envVarName: consts.OdigosEnvVarPodName}
+	errPodNameSpaceNotReported  = &errRequiredEnvVarNotFound{envVarName: consts.OdigosEnvVarNamespace}
+)
+
+type InstrumentationStatusReason string
+
+const (
+	FailedToLoad       InstrumentationStatusReason = "FailedToLoad"
+	FailedToInitialize InstrumentationStatusReason = "FailedToInitialize"
+	LoadedSuccessfully InstrumentationStatusReason = "LoadedSuccessfully"
+	FailedToRun        InstrumentationStatusReason = "FailedToRun"
+)
+
+type InstrumentationHealth bool
+
+const (
+	InstrumentationHealthy   InstrumentationHealth = true
+	InstrumentationUnhealthy InstrumentationHealth = false
+)
+
+type configUpdate struct {
+	workloadKey types.NamespacedName
+	config      *odigosv1.InstrumentationConfig
+}
+
+type ConfigUpdateFunc func(ctx context.Context, workloadKey types.NamespacedName, config *odigosv1.InstrumentationConfig) error
+
+type instrumentationDetails struct {
+	inst     Instrumentation
+	pod      types.NamespacedName
+	lang     common.ProgrammingLanguage
+	workload types.NamespacedName
+}
+
+type Manager struct {
+	// channel for receiving process events,
+	// used to detect new processes and process exits, and handle their instrumentation accordingly.
+	procEvents        <-chan detector.ProcessEvent
+	client            client.Client
+	factories         map[FactoryID]Factory
+	logger            logr.Logger
+
+	// all the active instrumentations by pid,
+	// this map is not concurrent safe, so it should be accessed only from the main event loop
+	detailsByPid      map[int]*instrumentationDetails
+
+	// active instrumentations by workload, and aggregated by pid
+	// this map is not concurrent safe, so it should be accessed only from the main event loop
+	detailsByWorkload map[types.NamespacedName]map[int]*instrumentationDetails
```

Comment:
add workload kind and language here and simplify the usage

Reply:
Done, PTAL